OpenSim/Common/ComponentConnector.h

```diff
@@ -344,36 +381,99 @@ class Connector : public AbstractConnector {
     }
 
     SimTK_DOWNCAST(Connector, AbstractConnector);
+    
+    /** For use in python/java/MATLAB bindings. */
+    // This method exists for consistency with Object's safeDownCast.
+    static Connector<T>* safeDownCast(AbstractConnector* base) {
+        return dynamic_cast<Connector<T>*>(base);
+    }
+    
+protected:
+    /** Create a Connector that can only connect to Object of type T with 
+    specified name and stage at which it should be connected. Only Component
+    should ever construct this class.
+    @param name               name of the connector used to describe its dependency.
+    @param connecteeNameIndex Index of the property in the containing Component
+                              that holds this Connector's connectee_name(s).
+    @param connectAtStage     Stage at which Connector should be connected.
+    @param owner              The component that contains this input. */
+    Connector(const std::string& name, const PropertyIndex& connecteeNameIndex,
+              const SimTK::Stage& connectAtStage,
+              Component& owner) :
+        AbstractConnector(name, connecteeNameIndex, connectAtStage, owner),
+        connectee(nullptr) {}
+        
+    /** So that Component can construct a Connector. */
+    friend Component;
 
 private:
     mutable SimTK::ReferencePtr<const T> connectee;
 }; // END class Connector<T>
             
 
 /** A specialized Connector that connects to an Output signal is an Input.
-    An AbstractInput enables maintenance of a list of unconnected Inputs. 
-    An Input can either be a single-value Input or a list Input. A list Input
-    can connect to multiple (Output) Channels.
+An AbstractInput enables maintenance of a list of unconnected Inputs. 
+An Input can either be a single-value Input or a list Input. A list Input
+can connect to multiple (Output) Channels.
+
+#### XML Syntax of a connectee name
+
+For every %Input that a component has, the XML representation of the component
+contains an element named `input_<input_name>_connectee_name` (or
+`input_<input_name>_connectee_names` for list inputs). For example, a component
+that has an Input named `desired_angle` might look like the following in XML:
+@code
+    <MyComponent name="my_comp">
+        <input_desired_angle_connectee_name>
+            ../foo/angle
+        </input_desired_angle_connectee_name>
+        ...
+    </MyComponent>
+@endcode
+You use this field to specify the outputs/channels that should be connected to
+this input (that is, the connectees). The syntax for the connectee name
+property is as follows:
+@code
+<path/to/component/><output_name>[:<channel_name>][(<alias>)]
```

Comment:
So, I actually tried again and went through with the change. The component path and the output name are now separated by `|`. I decided it would be good to have this change in this PR, to avoid introducing an excess XML format update.